CMakeLists.txt

```diff
@@ -0,0 +1,112 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+project(MerryMem
+    VERSION 0.1.0
+    LANGUAGES CXX
+)
+
+# Use C++20 without compiler-specific extensions.
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+set(CMAKE_CXX_EXTENSIONS FALSE)
+
+# Enable LTO for Release builds, if available.
+include(CheckIPOSupported)
+check_ipo_supported(RESULT MM_IPO_SUPPORTED)
+message(STATUS "IPO supported: ${MM_IPO_SUPPORTED}")
+if(MM_IPO_SUPPORTED)
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_DEBUG FALSE)
+endif()
```

Comment:
Wouldn't inter-procedural optimizations be expensive?

I see this is for release builds only in the comment, but I couldn't tell that's true from just looking at the code.

Reply:
Ok I just remembered why I didn't wrap it with that specific check - multi-config generators don't use CMAKE_BUILD_TYPE.

Tbh idk if anyone out there is using multi-config generators; I'm inclined to just not support it because it seems like a pain according to "openenclave" issue 756 (I don't want to link it and have it notify them lol).

Reply:
> multi-config generators don't use CMAKE_BUILD_TYPE.

I'm not sure what that means

Reply:
That's aight. Let's just not worry about it.